parser/prism/Translator.cc

```diff
@@ -1348,12 +1348,38 @@ unique_ptr<parser::Node> Translator::patternTranslate(pm_node_t *node) {
         case PM_IN_NODE: { // An `in` pattern such as in a `case` statement, or as a standalone expression.
             auto inNode = reinterpret_cast<pm_in_node *>(node);
 
-            auto sorbetPattern = patternTranslate(inNode->pattern);
-
+            auto prismPattern = move(inNode->pattern);
             auto inlineIfSingle = true;
             auto statements = translateStatements(inNode->statements, inlineIfSingle);
 
-            return make_unique<parser::InPattern>(location, move(sorbetPattern), nullptr, move(statements));
+            if (prismPattern != nullptr &&
+                (PM_NODE_TYPE_P(prismPattern, PM_IF_NODE) || PM_NODE_TYPE_P(prismPattern, PM_UNLESS_NODE))) {
+                pm_statements_node *prismStatements = nullptr;
+                std::unique_ptr<parser::Node> sorbetGuard;
+
+                if (PM_NODE_TYPE_P(prismPattern, PM_IF_NODE)) {
+                    auto ifNode = reinterpret_cast<pm_if_node *>(prismPattern);
+                    prismStatements = reinterpret_cast<pm_statements_node *>(ifNode->statements);
+                    sorbetGuard = std::make_unique<parser::IfGuard>(location, translate(ifNode->predicate));
+                } else { // PM_UNLESS_NODE
+                    auto unlessNode = reinterpret_cast<pm_unless_node *>(prismPattern);
+                    prismStatements = reinterpret_cast<pm_statements_node *>(unlessNode->statements);
+                    sorbetGuard = std::make_unique<parser::UnlessGuard>(location, translate(unlessNode->predicate));
+                }
+
+                if (prismStatements->body.size != 1) {
+                    unreachable("In pattern-matching's `in` clause, a guard must have a single statement.");
+                }
+
+                auto sorbetPattern = patternTranslate(prismStatements->body.nodes[0]);
+
+                return std::make_unique<parser::InPattern>(location, std::move(sorbetPattern), std::move(sorbetGuard),
```

Comment:
We have `using std::make_unique` and `using std::move`, so you can drop all these `std::` qualifiers

```suggestion
                return make_unique<InPattern>(location, move(sorbetPattern), move(sorbetGuard),
```